### PR TITLE
Global fixes for minor updates & corrections

### LINKED ISSUE #135: Revise Description to Meet GDPR Compliance - REMOVAL OF  " Location-based advertising" in Documentation
### Problem Description
**Remove "Location-based advertising" from YAML.** 

- “Location Verification could be useful in scenarios such as... Location-based advertising, to trigger _**targeted advertising_** after verifying the user is in the area of interest...” 

The API’s reference to location-based advertising is problematic for GDPR. There’s uncertainty about whether users have agreed to let their location data be used for _**targeted advertising**_.

### **Expected Action**

I strongly urge the group to remove that specific use case from the YAML, "Location-based advertising, to trigger targeted advertising after verifying the user is in the area of interest" from the API documentation. 

### **Additional Context**
 For GDPR this API does not meet the standards of data ethics and legal compliance.

### GDPR concerns
- **Consent for Data Processing** Section 4(1) stipulates that personal data can only be processed with the consent of the Data Principal for lawful purposes. The device location API, as described, seems to lack a clear mechanism for obtaining user consent, particularly for the purpose of location-based **targeted advertising** especially since this is in essence granting consent through a back door approach. 

- **Potential for Unlawful Data Profiling and Automated Decision-Making** The use of data for automated decision-making or profiling without proper safeguards or consent contravenes Article 22 on automated individual decision-making, including profiling.

- **Non-Compliance with Children’s Data Protection:** If the API processes children’s data for advertising without proper consent, it would violate Article 8 concerning conditions applicable to child’s consent in relation to information society services rights.

### LINKED ISSUE #136: Correct YAML examples to use upper case enum of areaType
**Problem description**
To correct YAML examples to use upper case enum of areaType:
- CIRCLE

**Expected action**
Correction of the YAMLs to reflect the correct enum

**Additional context**
enum for areaType introduced in #132

### LINKED ISSUE #137: Fix format for attribute lastLocationTime in location retrieval
**Problem description**
In location-retrieval.yaml line 221, the `lastLocationTime` attribut format is `datetime` instead of `date-time`

**Expected behavior**
Fix format

**Alternative solution**

**Additional context**

### LINKED ISSUE #139: Rename in the OAS SubscriptionEventType to NotificationEventType
**Problem description**
Feedback from development team.
In the OAS:
line 469 we list the notification event type and not the subscription event type. In order to be clear for the developer better to rename this class `NotificationEventType`.
In line 459 with this modification we can rename `SubscriptionCreationEventType` to `SubscriptionEventType`

As the reminder we have distinct enum because `org.camaraproject.geofencing.v0.subscription-ends` is a notification type but not a subscription type.

We should have something like:
```
SubscriptionType:
  description: |
    area-entered - Event triggered when the device enters the given area
    area-left - Event triggered when the device leaves the given area
  type: string
  enum:
    - org.camaraproject.geofencing.v0.area-entered
    - org.camaraproject.geofencing.v0.area-left


NotificationType:
  description: |
    area-entered - Event triggered when the device enters the given area
    area-left - Event triggered when the device leaves the given area
    subscription-ends - Event triggered when the subscription ends
  type: string
  enum:
    - org.camaraproject.geofencing.v0.area-entered
    - org.camaraproject.geofencing.v0.area-left
    - org.camaraproject.geofencing.v0.subscription-ends
```

**Expected behavior**
Fix the swagger

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by bigludo7**
#### What type of PR is this?

Add one of the following kinds:
* bug
* correction

#### What this PR does / why we need it:

(all) Correct YAML examples to use upper case enum of areaType
(geofencing) Rename in the OAS SubscriptionEventType to NotificationEventType
(retrieval) Fix format for attribute lastLocationTime in location retrieval 
(retrieval & verification) Revise Description to Meet GDPR Compliance - REMOVAL OF " Location-based advertising" in Documentation (sentence from @iamdatanick  proposal)

#### Which issue(s) this PR fixes:

Fixes #136 
Fixes #139 
Fixes #137 
Fixes #135 

#### Changelog input

```
 Correct YAML examples to use upper case enum of areaType
-For geofencing: Rename in the OAS SubscriptionEventType to NotificationEventType
-For Retrieval:  Fix format for attribute lastLocationTime in location retrieval 
Revise Description to Meet GDPR Compliance - REMOVAL OF " Location-based advertising" in Documentation

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

File: code/API_definitions/location-retrieval.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ info:
 
     - Fraud protection to ensure a given user is located in the region, country or location authorized for financial transactions
     - Verify the GPS coordinates reported by the app on a device to ensure the GPS was not faked e.g. for content delivery with regional restrictions
-    - Location-based advertising: trigger targeted advertising after retrieving the area where the user is localized
+    - Contextual-based advertising, to trigger ** advertising** after verifying the device is in the area of interest
```
Comment (jlurien) [inline code/API_definitions/location-retrieval.yaml:26]:
```suggestion
    - Contextual-based advertising, to trigger advertising after verifying the device is in the area of interest

```

File: code/API_definitions/location-verification.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ info:
 
     - Fraud protection, to ensure a given user is located in the location area claimed for financial transactions.
     - Verification of GPS coordinates reported by the app on a device, to ensure the GPS was not faked, e.g. for content delivery with regional restrictions.
-    - Location-based advertising, to trigger targeted advertising after verifying the user is in the area of interest.
+    - Contextual-based advertising, to trigger ** advertising** after verifying the device is in the area of interest.
```
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:26]:
```suggestion
    - Contextual-based advertising, to trigger advertising after verifying the device is in the area of interest.
```

**Review by jlurien — DISMISSED** [review]
Minor corrections, LGTM

File: code/API_definitions/geofencing.yaml [hunk 3]
```diff
@@ -450,13 +450,13 @@ components:
...
         type:
-          $ref: "#/components/schemas/SubscriptionCreationEventType"
+          $ref: "#/components/schemas/SubcriptionEventType"
       required:
         - device
         - area
         - type
 
-    SubscriptionCreationEventType:
+    SubcriptionEventType:
```
Comment (akoshunyadi) [inline code/API_definitions/geofencing.yaml:459]: Typo: Sub**s**criptionEventType

**Review by bigludo7 — COMMENTED** [review]
Fixed SubscriptionEventType typo
Thanks @akoshunyadi

File: code/API_definitions/location-verification.yaml [hunk 4]
```diff
@@ -100,7 +100,7 @@ paths:
                       publicAddress: 123.234.1.2
                       publicPort: 1234
                   area:
-                    areaType: Circle
+                    areaType: CIRCLE
```
[the remark this answers is not in the document]
Comment (bigludo7) [reply code/API_definitions/location-verification.yaml:103]: Sorry @akoshunyadi I do not get it. You have fixed it?

[the remark this answers is not in the document]
Comment (akoshunyadi) [reply code/API_definitions/location-verification.yaml:103]: No, it's in the line 204, I couldn't comment there

**Comment by jlurien** [comment]
Should we create the rc2 now? or do we wait after the next meeting?

**Comment by akoshunyadi** [comment]
I think, if this PR is merged we can create the RC2

**Comment by bigludo7** [comment]
Fixed the line 204 - sorry Akos for the misunderstanding - for me once merge we can create RC2.

**Review by jlurien — APPROVED** [review]
LGTM

**Pull request merged**